addons/project/models/res_partner.py

```diff
@@ -61,3 +61,34 @@ def _create_portal_users(self):
                 'active': True,
             })
         return created_users
+
+    def action_view_tasks(self):
+        self.ensure_one()
+        action = self.env["ir.actions.actions"]._for_xml_id("project.action_view_all_task")
+        action['display_name'] = _("%(partner_name)s's Tasks", partner_name=self.name)
+        action['context'] = {
+            'default_partner_id': self.id,
+        }
+        views = []
+        for view in action['views']:
+            if view[1] == 'form':
+                views.append((self.env.ref('project.view_task_form_res_partner').id, view[1]))
+            elif view[1] == 'kanban':
+                views.append((self.env.ref('project.view_task_kanban_res_partner').id, view[1]))
+            else:
+                views.append(view)
+        action['views'] = views
+        all_child = self.with_context(active_test=False).search([('id', 'child_of', self.ids)])
+        if self.task_count <= 1:
+            task_id = self.env['project.task'].search([
+                '|',
+                    ('partner_id', 'in', self.ids),
+                    ('partner_id', 'in', all_child.ids),
+            ], limit=1)
+            action['res_id'] = task_id.id
+            action['views'] = [
+                (self.env.ref('project.view_task_form_res_partner').id, 'form'),
+            ]
+        else:
+            action['domain'] = ['|', ('partner_id', 'in', self.ids), ('partner_id', 'in', all_child.ids)]
```

Comment:
```suggestion
            action['domain'] = [('partner_id', 'in', (self | all_child).ids)]
```

Reply:
Maybe, this domain can be declared before the if statement, and used in the search (in the if block) or set in the action (here)?